atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CheckAndSetCompatibility.java

```diff
@@ -26,29 +26,53 @@
  * if so the granularity with which it can provide feedback.
  */
 public enum CheckAndSetCompatibility {
-    NOT_SUPPORTED,
     /**
-     * The {@link KeyValueService} supports CAS and PUE operations. However, in the event of failure, there are no
-     * guarantees that {@link CheckAndSetException#getActualValues()} or
-     * {@link KeyAlreadyExistsException#getExistingKeys()} actually return any meaningful data (other than the
-     * fact that the operation failed).
+     * This is the only value that is allowed if a {@link KeyValueService} does NOT support check and set.
      */
-    SUPPORTED_NO_DETAIL_ON_FAILURE,
+    NO_DETAIL_CONSISTENT_ON_FAILURE(false, true),
```

Comment:
Hmm, I really don't think we should use an enum here - we care more about whether something supports detail on failure, or whether something is consistent of failure (as opposed to which configuration it is).